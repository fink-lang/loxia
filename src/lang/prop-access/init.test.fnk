{fink2js} = import '../../testing/generate.fnk'
{describe, it, expect, to_match_snapshot} = import '@fink/jest/test.fnk'


describe 'member', fn:
  it 'compiles', fn:

    expect
      fink2js "
        foo = spam.shrub
        computed_member = item.'bar spam'
        computed_member2 = item.(get_key foo)
        reserved_prop = item.arguments
        js_safe_unicode = [item.π, item.ƒ]
        js_unsfae = item.foo-bar
      "
      to_match_snapshot

