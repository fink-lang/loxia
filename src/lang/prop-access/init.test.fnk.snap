// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`member compiles 1`] = `
"export const foo = spam.shrub;
export const computed_member = item[\`bar spam\`];
export const computed_member2 = item[get_key(foo)];
export const reserved_prop = item.arguments;
export const js_safe_unicode = [item.π, item.ƒ];
export const js_unsfae = item[\\"foo-bar\\"];"
`;
